pkg/package-server/storage/packagemanifest/watcher.go

```diff
@@ -0,0 +1,86 @@
+package packagemanifest
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/apis/packagemanifest/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/provider"
+)
+
+type Watcher struct {
+	namespace       string
+	name            string
+	resourceVersion string
+	labelSelector   labels.Selector
+
+	source provider.PackageManifestProvider
+
+	stop   chan struct{}
+	result chan watch.Event
+}
+
+var _ watch.Interface = &Watcher{}
+
+func NewWatcher(namespace, name, resourceVersion string, labelSelector labels.Selector, source provider.PackageManifestProvider) *Watcher {
+	return &Watcher{
+		namespace:       namespace,
+		resourceVersion: resourceVersion,
+		labelSelector:   labelSelector,
+		source:          source,
+		stop:            make(chan struct{}),
+		result:          make(chan watch.Event),
+	}
+}
+
+// Run is a blocking method which starts the watch by subscribing to the source's channels.
+// Should run in a goroutine.
+func (w *Watcher) Run() {
```

Comment:
I think you want to take a context.Context here

Reply:
@ecordell Passing and handling `context.Context` now.